homeassistant/components/rest/__init__.py

```diff
@@ -53,7 +54,7 @@
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the rest platforms."""
-    component = EntityComponent(_LOGGER, DOMAIN, hass)
+    component = EntityComponent[Entity](_LOGGER, DOMAIN, hass)
```

Comment:
I've used the base Entity class here because I wasn't quite sure.
Maybe I should be using `# type: ignore[var-annotated]` instead?

Reply:
Maybe this?
```suggestion
    component: EntityComponent = EntityComponent(_LOGGER, DOMAIN, hass)
```

As mentioned in a earlier comment, the mypy error is wrong. Just `EntityComponent(...)` shouldn't trigger `var-annotated`.